controllers/gcpmachine_controller_unit_test.go

```diff
@@ -81,23 +80,21 @@ func TestGCPMachineReconciler_GCPClusterToGCPMachines(t *testing.T) {
 		newMachine(clusterName, "my-machine-2"),
 	}
 
-	client := fake.NewFakeClientWithScheme(scheme, initObjects...)
+	client := fake.NewClientBuilder().WithScheme(scheme).WithRuntimeObjects(initObjects...).Build()
 
 	reconciler := &GCPMachineReconciler{
 		Client: client,
 		Log:    klogr.New(),
 	}
-	requests := reconciler.GCPClusterToGCPMachines(handler.MapObject{
-		Object: &infrav1.GCPCluster{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      clusterName,
-				Namespace: "default",
-				OwnerReferences: []metav1.OwnerReference{
-					{
-						Name:       clusterName,
-						Kind:       "Cluster",
-						APIVersion: clusterv1.GroupVersion.String(),
-					},
+	requests := reconciler.GCPClusterToGCPMachines(&infrav1.GCPCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      clusterName,
+			Namespace: "default",
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					Name:       clusterName,
+					Kind:       "Cluster",
+					APIVersion: clusterv1.GroupVersion.String(),
 				},
 			},
 		},
```

Comment:
Can't comment on the specific line, but wondering if this test should also do additional verification of the contents of the results instead of just the length of results, similar to what is being done in `TestDockerMachineReconciler_DockerClusterToDockerMachines()` here: https://github.com/kubernetes-sigs/cluster-api/blob/master/test/infrastructure/docker/controllers/dockermachine_controller_test.go#L56